qutip/solver/dysolve_propagator.py

```diff
@@ -254,32 +249,31 @@ def _update_matrix_elements(self, current: ArrayLike) -> ArrayLike:
         current : ArrayLike
             The current matrix elements (for the order n-1).
 
+        elems : ArrayLike
+            The matrix elements of self.X.
+
         Returns
         -------
         matrix_elements : ArrayLike
             The new matrix elements for the order n.
         """
-        elems = self._X.transform(self._basis).full().flatten()
         if current is None:
             return elems
         else:
             shape = self._X.shape[0]
             a = np.tile(current, shape)
             b = np.repeat(elems, len(current)//shape)
             return a * b
-
-        # elems = sp.sparse.csr_array(
-        #     self.X.transform(self._basis).full().flatten()
-        # )
+        
         # if current is None:
         #     return elems
         # else:
-        #     x_shape = self.X.shape[0]
+        #     x_shape = self._X.shape[0]
```

Comment:
Let's remove unused code or add comment as to why they are there and how to use them in the future if they are about features yet to be implemented.